clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/util/DeployDcosServerGroupDescriptionToAppMapper.groovy

```diff
@@ -124,28 +126,37 @@ class DeployDcosServerGroupDescriptionToAppMapper {
         parsedConstraints
     }
 
-    public Map<String, String> parsePortMappingLabels(String appId, List<DeployDcosServerGroupDescription.ServiceEndpoint> serviceEndpoints) {
-        Map<String, String> parsedMap = [:]
+    private Tuple parsePortMappingLabels(String appId, DeployDcosServerGroupDescription.ServiceEndpoint serviceEndpoint, int counter) {
+        Map<String, String> parsedLabels = serviceEndpoint.labels.clone()
 
-        int counter = 0
+        List<String> vipKeys = parsedLabels.keySet().stream().filter({ key -> key.matches(VIP_PATTERN) })
+                .collect(Collectors.toList())
 
-        serviceEndpoints.stream().filter({
-            serviceEndpoint -> serviceEndpoint.isLoadBalanced
-        }).forEach({
-            serviceEndpoint -> parsedMap.put("VIP_${counter++}", "${appId}:${serviceEndpoint.port}")
+        vipKeys.stream().forEach({
+            key ->
+                String vip = parsedLabels.remove(key)
+                parsedLabels.put("VIP_${counter++}".toString(), vip)
         })
 
-        parsedMap
+        if (serviceEndpoint.loadBalanced && vipKeys.isEmpty()) {
+            parsedLabels.put("VIP_${counter++}".toString(), "${appId}:${serviceEndpoint.port}".toString())
+        }
+
+        return [counter, parsedLabels]
     }
 
     public List<Port> parsePortMappings(String appId, List<DeployDcosServerGroupDescription.ServiceEndpoint> serviceEndpoints) {
+
+        int counter = 0
```

Comment:
Is there ever a case where an endpoint should have multiple VIPs? I don't feel like we really should need that.